viewing_party/party.py

```diff
@@ -0,0 +1,134 @@
+            genre_frequencies[user_data["watched"][i]["genre"]] = 1
+        else:
+            genre_frequencies[user_data["watched"][i]["genre"]] += 1
+    if genre_frequencies != {}:
+        most_watched_genre = max(genre_frequencies, key = lambda genre : genre_frequencies[genre])
+        return most_watched_genre
+    else:
+        return None
+
+# wave03
+
+def get_unique_watched(user_data):
+    # determines which movies the user has seen that friends have not and returns a list of them
+    all_friends_watched = get_all_friends_watched_no_duplicates(user_data)
+    in_common_watched = []
+    unique_watched = []
+    # makes a list of the movie dicts the friends have watched that user has also watched
+    for i in range(len(user_data["watched"])):
+        for j in range(len(all_friends_watched)):
+            if user_data["watched"][i]["title"] == all_friends_watched[j]["title"]:
+                in_common_watched.append(user_data["watched"][i])
+    # removes the movie dicts friends and user have in common
+    unique_watched = user_data["watched"]
+    for movie in in_common_watched:
+        unique_watched.remove(movie)
+    return unique_watched
+    
+def get_friends_unique_watched(user_data):
+    all_friends_watched = get_all_friends_watched_no_duplicates(user_data)
+    in_common_watched = []
+    friend_unique_watched = []
+    # makes a list of the movie dicts the friends have watched that user has also watched
+    for i in range(len(user_data["watched"])):
+        for j in range(len(all_friends_watched)):
+            if user_data["watched"][i]["title"] == all_friends_watched[j]["title"]:
+                in_common_watched.append(all_friends_watched[j])
+    # removes the movie dicts friends and user have in common  
+    friend_unique_watched = all_friends_watched
+    for movie in in_common_watched:
+        friend_unique_watched.remove(movie)
+    return friend_unique_watched
+
+# wave04
+
+def get_available_recs(user_data):
+    available_recs = []
+    if get_friends_unique_watched(user_data):
+        all_recs = get_friends_unique_watched(user_data)
+    for movie in all_recs:
+        if movie["host"] in user_data["subscriptions"]:
+            available_recs.append(movie)
+    return available_recs
+
+# wave 05
+
+def get_new_rec_by_genre(user_data):
+    most_watched_genre = get_most_watched_genre(user_data)
+    possible_recs = get_friends_unique_watched(user_data)
+    recs_by_genre = []
+    for movie in possible_recs:
+        if movie["genre"] == most_watched_genre:
+            recs_by_genre.append(movie)
+    return recs_by_genre
+
+def get_rec_from_favorites(user_data):
+    recs_from_favorites = []
+    all_friends_watched = get_all_friends_watched_no_duplicates(user_data)
+    for i in range(len(user_data["favorites"])):
+        if user_data["favorites"][i] not in all_friends_watched:
+            recs_from_favorites.append(user_data["favorites"][i])
```

Comment:
One more opportunity to rephrase using a loop over the items directly rather than the position. But otherise, looks good!